### ISSUE TITLE
VDI without VBD causes silent failure to create VM

### ISSUE THREAD

**Issue by briantopping** [ringods/terraform-provider-xenserver#21]
If a template has a VDIs referencing it that do not have VBDs, the create script silently fails. 

It appears that https://github.com/ringods/terraform-provider-xenserver/blob/c77ef9919202fd201992f7680469ba173515cacb/xenserver/resource_vbd.go#L224 should handle this better.

I'm just learning Go, so appreciate the opportunity to get involved with this fix. Just wanted to report it to get started.

**Comment by briantopping** [comment]
On second pass, it looks like there was an extra VBD for the template. I guess I haven't been studying XenAPI much lately. Once that was deleted from the template, things worked properly.

**Comment by ringods** [comment]
@briantopping I still have to learn a lot myself too. 😄 

Can I close this issue then?

**Comment by briantopping** [comment]
Hi @ringods, there's really two problems here, one is that it can't handle bad templates (which is pretty reasonable), but I'd like to see it fail with an error instead of give indication at the command line that everything worked. WDYT?

**Comment by mborodin** [comment]
Hi @briantopping, can you provide a more detailed example? The line you are referencing here is just a query to XenServer to obtain info on particular VBD.

**Comment by briantopping** [comment]
Hi @mborodin, thanks for the note. Yes, it does appear quite benign!

I was unfortunately unable to reproduce the broken VBD/VDI and provide instructions, but it was a strange case where the call to load the VDI would fail. Looking at the VBD in the `xe` CLI, the VDI showed as empty.

In studying this some, I came to realize the problem and created a patch. `SetID` was being called as soon as the template was cloned. https://www.terraform.io/guides/writing-custom-terraform-providers.html#error-handling-amp-partial-state provides a nuanced list of how TF reacts to resource creation. If an ID is set, regardless of whether an error occurred, the resource is assumed to be created. 

In our case, we have created some parts of the resource, specifically the clone of the template could cause errors if the `apply` was run again. So I also added in this patch calls to `Partial`. I believe if the `apply` was run again with this new code, it would not fail on the template being cloned because the partial state? I'm not exactly sure, but I believe that's the case.
